File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/response/GetProblemsResponse.java
package com.github.ai.leetcodequiz.api.response;

import com.github.ai.leetcodequiz.api.ProblemsItemDto;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record GetProblemsResponse(
    @NotNull List<ProblemsItemDto> problems
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/response/GetProblemsResponse.scala
package com.github.ai.leetcodequiz.api.response

import com.github.ai.leetcodequiz.api.ProblemsItemDto
import zio.json.{JsonDecoder, JsonEncoder}

case class GetProblemsResponse(
  problems: List[ProblemsItemDto]
) derives JsonEncoder, JsonDecoder
